### ISSUE TITLE
CA2254 Is non-functional with Microsoft.Azure.WebJobs

### ISSUE THREAD

**Issue by philipborg** [dotnet/roslyn-analyzers#5987]
### Analyzer

**Diagnostic ID**: [CA2254](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2254): `Template should be a static expression`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.202](https://dotnet.microsoft.com/download/dotnet/6.0)

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest) and 7.0.0-preview1.22168.1 (Preview)

### Describe the bug

CA2254 seemingly never gives any warnings even if all other rules seem to work if `Microsoft.Azure.WebJobs` version `3.0.32` is installed. I have tried both the built-in CA analyzer, NuGet 6.0.0 and 7.0.0-preview1.22168.1, never works.  I have also tried lowering `Microsoft.Azure.WebJobs` to `3.0.0` without success.

Tried properties:

```
<EnableNETAnalyzers>true</EnableNETAnalyzers>
<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
<AnalysisLevel>latest-all</AnalysisLevel>
```

I have also tried to set AnalysisLevel to `6-all` and `preview-all`, with and without nuget package. Same issue persists.

### Steps To Reproduce

1. Enable CA2254 analyzer.
2. Add the following code.
```csharp
using Microsoft.Extensions.Logging;

namespace CA2254POC;

public class Sample
{
    public void LogExample(ILogger<Sample> logger)
    {
        logger.LogInformation($"{logger.GetType().Assembly.Location}");
    }
}
```
Sample csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <TargetFramework>net6.0</TargetFramework>
        <ImplicitUsings>enable</ImplicitUsings>
        <LangVersion>latest</LangVersion>
        <Nullable>enable</Nullable>
        <EnableNETAnalyzers>true</EnableNETAnalyzers>
        <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
        <AnalysisLevel>latest-all</AnalysisLevel>
    </PropertyGroup>

    <ItemGroup>
      <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="6.0.1" />
    </ItemGroup>
</Project>
```
3. It will flag this sample code correctly.
4. Add dependency to `Microsoft.Azure.WebJobs`
```xml
    <ItemGroup>
        <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.32" />
    </ItemGroup>
```
5. CA2254 analyzer now always returns negative responses.

### Expected behavior

CA2254 should trigger even with `Microsoft.Azure.WebJobs` installed.

### Actual behavior

CA2254 doesn't trigger with `Microsoft.Azure.WebJobs` installed.

## Additional context

I have tried just having each of the dependencies of `Microsoft.Azure.WebJobs` installed and the problem did not occur. I am thus pretty certain it's a problem with `Microsoft.Azure.WebJobs` specifically rather than any one of its dependencies. I initially discovered this issue when `Microsoft.NET.Sdk.Functions` `4.1.0` or `4.0.0` was installed and boiled it down to the `Microsoft.Azure.WebJobs` dependency.

**Comment by Bellarmine-Head** [comment]
What this means (just one example) is that this warning will never trigger for Azure Function projects.

**Comment by Bellarmine-Head** [comment]
It would seem that https://docs.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca2017 is also broken in `Azure Function` projects.

**Comment by remoba** [comment]
Still broken in the latest Microsoft.Azure.WebJobs version (3.0.37), any insights to share? At least an acknowledgement that this is a bug?

**Comment by dmeytin** [comment]
+1 on fixing this bug
